### PR TITLE
core: Add a config option "--with-lttng"

### PR DESCRIPTION
**Pull request by jtamzn**
This PR add an option to enable userspace tracing capability for EFA. This PR only adds a macro definition, while exact tracepoints will be defined later.

Signed-off-by: Tang, Jingyin <jytang@amazon.com>

### CODE REVIEW

File: configure.ac
Comment (wzamazon) [inline configure.ac:697]: use `FI_CHECK_PACKAGE` to check the existence of lttng library

**Comment by wzamazon** [comment]
The commit message is wrong, this change is for core, not just for EFA.

**Review by wzamazon — CHANGES_REQUESTED** [review]
The first commit is a core change, and need community approval

The second commit is efa change (that rely on 1st commit). Can you remove it from this PR? and wait until the core change is merged to open it?

File: configure.ac
Comment (shefty) [reply configure.ac:697]: Update the commit message to reference configure or build or core, and remove the reference to prov/efa, as these changes are not limited to efa.

File: prov/efa/src/rxr/rxr_pkt_type_misc.c
Comment (shefty) [inline prov/efa/src/rxr/rxr_pkt_type_misc.c:416]: I know this is in the efa provider, so I don't really care, but the line lengths here are getting ridiculous.

File: prov/efa/src/rxr/efa_tp_def.h
Comment (shefty) [inline prov/efa/src/rxr/efa_tp_def.h:3]: This patch looks like it should have been merged with the previous patch.  Don't patch a patch that hasn't been merged.  Just update the original patch directly.

Comment (jtamzn) [reply prov/efa/src/rxr/efa_tp_def.h]: I need clean up patches around. Wei commented some changes will cause additional changes around. I will get them in one by one.

**Comment by jtamzn** [comment]
I am going to cancel this request and resend a new one. Things are getting messed. Apologize for any inconvenience.

File: configure.ac
Comment (wzamazon) [inline configure.ac:29]: Did you replace tab with space?

**Pull request closed without merging**
